docs/arch/agent.md

```diff
@@ -0,0 +1,67 @@
+
+# Summary
+The agent is the powerhouse of SealCi. It will receive jobs and try to run them in order to complete the operational part of the CI.
+
+# The role of the agent
+The goal of this component  is to provide execution environment for the scheduler and safe and correctly configured environments.
+
+# Agent operations
+
+## Registering to the scheduler
+The agent is a slave of the scheduler and therefore he needs to be able to communicate with it. The agent will register to the scheduler at its startup. The scheduler will then be able to send jobs to the agent.
+
+## Health link
+At any moment in its life, the agent must be able to provide information about its health status. This information will be sent to the scheduler so that it can make decisions accordingly.
+
+TODO: parler de la socket ouverte pour envoyer les infos de santé en continu
+
+## Getting jobs
+Jobs are the composition of configuration for the execution environnement and for the execution itself that will run in an encapsulated environment. The agent must have access to this environment in order to manage it.
+The scheduler will send jobs to the agent for it to execute them taking into account the health status of the agent, 
+
+TODO: changer pour dire que le scheduler ouvre une socket pour envoyer le info sur un job et qu'il y a donc une socket par job, et que si la socket est fermée le job est annulé.
+
+## Creating encapsulated environment
+The agent will try to create an encapsulated environment from a given configuration. This mechanism is mandatory, if it fail the entire job will fail. 
+If it successful, the job will process the execution configuration and run it. 
+
+The agent must have access to every logs of the job both from the creation of the environment and execution part. To be able to stream them to the scheduler.
+
+## Running jobs
+
+Once the environment has been created and configured and the job has been received, the agent must execute the job.
```

Comment:
[scheduler team]
Does the received job contain the environment ocnfiguration ?

My first thought is that there is a default configuration, and the Job may contain overloading environment configuration data such as (but not limited to?) environment variables ?

So the environment could be configured through scripts ?

Reply:
As we go in the project we think we will use containers to manage those processes. So the configuration should be the image the container will use.